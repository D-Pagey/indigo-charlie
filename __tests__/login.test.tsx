import "@testing-library/jest-dom";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Login from "../pages/login";

const mockRouterPush = jest.fn();

jest.mock("next/router", () => ({
  useRouter: () => ({
    push: mockRouterPush,
  }),
}));

describe("Login", () => {
  it("renders", () => {
    const { container } = render(<Login />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should show errors if no username or password provided", async () => {
    const user = userEvent.setup();

    render(<Login />);

    const loginButton = screen.getByRole("button", { name: "Login" });

    await user.click(loginButton);

    expect(screen.getByText("A username is required")).toBeInTheDocument();
    expect(screen.getByText("A password is required")).toBeInTheDocument();
  });

  it("should show password required if username but no password provided", async () => {
    const user = userEvent.setup();

    render(<Login />);

    const loginButton = screen.getByRole("button", { name: "Login" });
    const usernameInput = screen.getByLabelText("Username");

    await user.type(usernameInput, "incard");
    await user.click(loginButton);

    expect(screen.getByText("A password is required")).toBeInTheDocument();
  });

  it("should show username required if password but no username provided", async () => {
    const user = userEvent.setup();

    render(<Login />);

    const loginButton = screen.getByRole("button", { name: "Login" });
    const passwordInput = screen.getByLabelText("Password");

    await user.type(passwordInput, "incard");
    await user.click(loginButton);

    expect(screen.getByText("A username is required")).toBeInTheDocument();
  });
});
