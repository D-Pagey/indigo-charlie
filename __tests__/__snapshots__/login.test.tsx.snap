// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<main>
  <h1
    class="text-3xl mb-4"
  >
    Login
  </h1>
  <form
    class="flex flex-col text-black"
  >
    <label
      class="flex flex-col text-white mb-4"
    >
      Username
      <input
        class="mt-2 p-2 rounded text-lg text-black"
        name="username"
        placeholder="Enter your username..."
      />
    </label>
    <label
      class="flex flex-col text-white mb-4"
    >
      Password
      <input
        class="mt-2 p-2 rounded text-lg text-black"
        name="password"
        placeholder="Enter your password..."
      />
    </label>
    <input
      class="text-white bg-primary p-2 rounded-lg mt-4 cursor-pointer"
      type="submit"
      value="Login"
    />
  </form>
</main>
`;
